README
SYNOPSIS

        In the dist.ini:
    
        [ChangeStats::Git]
        group=STATISTICS
        skip_match=^meta

DESCRIPTION

    Adds a line to the changelog giving some stats about the code churn
    since the last release, which will look like:

      - code churn: 6 files changed, 111 insertions(+), 1 deletions(-)

ARGUMENTS

 group

    If given, the line is added to the specified group.

 develop_branch

    The master developing branch. Defaults to master.

 auto_previous_tag

    If enabled, look in the guts of the Dist::Zilla::Plugin::Git::Tag
    plugin in order to find the previous release's tag. This will be then
    compared against the develop_branch. Defaults to false (0).

 release_branch

    The branch recording the releases. Defaults to releases.

 text

    The text before git output. If it is a non-empty string, :<space> will
    be appended. Defaults to code churn.

 skip_file

    A complete path (from the distribution root) that should not be
    included in the statistics. Can be given multiple times.

 skip_match

    A part of a regex used to match against complete paths. Paths that
    match are not included in the statistics. Can be given multiple times.



Changes
Revision history for Dist-Zilla-Plugin-ChangeStats-Git

{{$NEXT}}
  [ API CHANGES ]

  [ BUG FIXES ]

  [ DOCUMENTATION ]

  [ ENHANCEMENTS ]

  [ NEW FEATURES ]

  [ STATISTICS ]

0.6.0 2017-04-14
  [ ENHANCEMENTS ]
    - plugin doesn't crash if the git branches used for the diff don't
      exist.

  [ STATISTICS ]
    - code churn: 3 files changed, 82 insertions(+), 69 deletions(-)

0.5.1 2016-02-22
  [ BUG FIXES ]
    - Fix deletions count bug when skipping files. (GH#7, Erik Carlsson)

  [ STATISTICS ]
    - code churn: 2 files changed, 10 insertions(+), 3 deletions(-)

0.5.0 2016-02-13
  [ ENHANCEMENTS ]
    - new plugin options 'text', 'skip_file', and 'skip_match'.  (GH#5,
      Erik Carlsson)

  [ STATISTICS ]
    - code churn: 5 files changed, 241 insertions(+), 76 deletions(-)

0.4.0 2015-10-01
  [ BUG FIXES ]
    - Fix regex warnings (and tests) in recent perls. (GH#4, Shlomi Fish)

  [ ENHANCEMENTS ]
    - Allow to retrieve last version from Git::Tag. (GH#3, apocalypse)

  [ STATISTICS ]
    - code churn: 5 files changed, 62 insertions(+), 14 deletions(-)

0.3.0 2014-09-09
  [ ENHANCEMENTS ]
    - Can configure develop and releases branches. (GH#2)

  [ STATISTICS ]
    - code churn: 3 files changed, 35 insertions(+), 3 deletions(-)

0.2.1 2013-05-02
  [ BUG FIXES ]
    - Don't drop statistics in NEXT release in the on-disk changelog.

  [ STATISTICS ]
    - code churn: 2 files changed, 15 insertions(+), 5 deletions(-)

0.2.0 2013-05-02
  [ ENHANCEMENTS ]
    - Also write the added [STATISTICS] to the on-disk changelog.

  [ STATISTICS ]
    - code churn: 5 files changed, 50 insertions(+), 8 deletions(-)

0.1.3 2012-06-03
  [ BUG FIXES ]
    - Three dots, for heaven's sake! (releases...master)

  [ STATISTICS ]
    - code churn: 18 files changed, 82 insertions(+), 1400 deletions(-)

0.1.2 2012-06-03
  [ BUG FIXES ]
    - Stats taken from 'git diff', not 'git log'.

0.1.1 2012-05-22
  [ BUG FIXES ]
    - I got the pre/post munging all wrong

0.1.0 2012-05-22
  - First version, unleashed on an unsuspecting world.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.28"
  },
  "configure_requires" => {
    "Module::Build" => "0.28"
  },
  "dist_abstract" => "add code churn statistics to the changelog",
  "dist_author" => [
    "Yanick Champoux <yanick\@babyl.dyndns.org>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-ChangeStats-Git",
  "dist_version" => "0.3.0",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::ChangeStats::Git",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "CPAN::Changes" => "0.17",
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Git::Repository" => 0,
    "List::Util" => 0,
    "Module::Load" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "Moose::Util" => 0,
    "MooseX::Role::Parameterized" => 0,
    "Path::Tiny" => 0,
    "Perl::Version" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.88",
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Module::Build" => "0.28",
  "Test::DZil" => 0,
  "Test::More" => "0.88",
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "add code churn statistics to the changelog",
  "AUTHOR" => "Yanick Champoux <yanick\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-ChangeStats-Git",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::ChangeStats::Git",
  "PREREQ_PM" => {
    "CPAN::Changes" => "0.17",
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::FileMunger" => 0,
    "Dist::Zilla::Role::Plugin" => 0,
    "Git::Repository" => 0,
    "List::Util" => 0,
    "Module::Load" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "Moose::Util" => 0,
    "MooseX::Role::Parameterized" => 0,
    "Path::Tiny" => 0,
    "Perl::Version" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.5.1",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "CPAN::Changes" => "0.17",
  "Dist::Zilla::Role::AfterRelease" => 0,
  "Dist::Zilla::Role::FileMunger" => 0,
  "Dist::Zilla::Role::Plugin" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Git::Repository" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "List::Util" => 0,
  "Module::Load" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "Moose::Util" => 0,
  "MooseX::Role::Parameterized" => 0,
  "Path::Tiny" => 0,
  "Perl::Version" => 0,
  "Test::DZil" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
